File: app/src/main/java/com/xiamen/www/myapplication3/kotlin/javatest/design_pattern/builder/protetype/Person.java
package com.xiamen.www.myapplication3.kotlin.javatest.design_pattern.builder.protetype;

/**
 * Created by White on 2018/4/27.
 */

public class Person {
    private String name;
    private boolean sex;
    private int age;
    private float height;
    private float weight;


    public Person(String name, boolean sex, int age, float height, float weight) {
        this.name = name;
        this.sex = sex;
        this.age = age;
        this.height = height;
        this.weight = weight;
    }
}

File: app/src/main/java/com/xiamen/www/myapplication3/kotlin/module/callback/Person.kt
package com.xiamen.www.myapplication3.kotlin.module.callback

/**
 * Created by White on 2018/3/21.
 */
class Person {

    val name: String = "Person"
    lateinit var mListen: MyInterface

    fun setListener(listen: MyInterface) {
        this.mListen = listen
        this.mListen.poo("poo" + name)
    }

    interface MyInterface {
        fun poo(str: String)
    }
}


fun main(args: Array<String>) {
    println("Hello world")
    val person = Person()
    person.setListener(object : Person.MyInterface {
        override fun poo(str: String) {
            println(str)
        }

    })
}